go/vt/vtgate/engine/comparer.go

```diff
@@ -23,16 +23,22 @@ import (
 
 // comparer is the struct that has the logic for comparing two rows in the result set
 type comparer struct {
-	orderBy, weightString int
-	desc                  bool
+	orderBy, weightString, starColFixedIndex int
```

Comment:
can we avoid storing `starColFixedIndex` and just use `orderBy`?

Reply:
some place also use orderBy to handle some logic, like "select t.*, t.id from t order by t.id",  there have some logic to handle t.id to save index 1, and other place get the map["1"] to get the orderBy. 
So I use another var to avoid error.